firedrake/mesh.py

```diff
@@ -2366,6 +2397,44 @@ def _pic_swarm_in_plex(plex, coords, fields=[]):
     return swarm
 
 
+def _parent_mesh_embedding(vertex_coords, parent_mesh):
```

Comment:
This needs moving to cython. It all looks quite similar to stuff I've got there already which is handy

Reply:
I don't think this is necessary. The code which is in cython immediately calls back to plain python anyway so I don't see what we get by being in cython.

Reply:
I was merely thinking about the loop which, in many cases, will be over large numbers of vertices. Happy to be educated though

Reply:
The call back to Python is inside that loop. That's why I think cython doesn't gain us anything here.

Reply:
Well nevertheless (and this would be a fun thing to profile at some point) this seems to be very similar to this function: https://github.com/firedrakeproject/firedrake/blob/9c9bc952e082f066b0f9ffe547e02d95fb4d78c9/firedrake/cython/dmcommon.pyx#L2907-L2980

Reply:
We discussed this offline. Both of these functions almost certainly do not gain from being in cython.